server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1457,7 +1457,7 @@ protected VolumeVO retrieveAndValidateVolume(long volumeId, Account caller) {
      * The volume is destroyed via {@link VolumeService#destroyVolume(long)} method.
      */
     protected void destroyVolumeIfPossible(VolumeVO volume) {
-        if (volume.getState() != Volume.State.Destroy && volume.getState() != Volume.State.Expunging && volume.getState() != Volume.State.Expunged && volume.getState() != Volume.State.Allocated && volume.getState() != Volume.State.Uploaded) {
+        if (volume.getState() != Volume.State.Destroy && volume.getState() != Volume.State.Expunging && volume.getState() != Volume.State.Expunged && volume.getState() != Volume.State.Allocated) {
```

Comment:
1. replace big "IFs" to a Set of the ENUM items that should (or not) match with the compared value.
```
        private static final Set<Volume.State> STATES_VOLUME_CANNOT_BE_DESTROYED = new HashSet<>(Arrays.asList(Volume.State.Destroy, Volume.State.Expunging, Volume.State.Expunged, Volume.State.Allocated));
...
        if (!STATES_VOLUME_CANNOT_BE_DESTROYED.contains(volume.getState())) {
```


2. the other suggestion would be to update the Javadoc by adding the Volume.State.Allocated

```
     * <ul>
     *  <li> {@value Volume.State#Destroy};
     *  <li> {@value Volume.State#Expunging};
     *  <li> {@value Volume.State#Expunged}.
     * </ul>
```

Reply:
Thanks for the review @GabrielBrascher. I've addressed your comments.